### ISSUE TITLE
codes.Internal returned for context.DeadlineExceeded on client

### ISSUE THREAD

**Issue by maxhawkins** [grpc/grpc-go#129]
I'm making a client-side RPC call to a grpc-go server that has crashed. My context has a timeout (context.WithTimeout) so after a few seconds the RPC fails as expected with the following message:

```
2015/03/22 14:01:04 [error] rpc error: code = 13 desc = "stream error: code = 4 desc = \"context deadline exceeded\""
```

However, the grpc.Code for this error is codes.Internal rather than codes.DeadlineExceeded.

I don't know the codebase well enough to add a failing test case, however I believe the issue comes from this line, which returns all errors with the code codes.Internal:
https://github.com/grpc/grpc-go/blob/fbd3f79d38f568d15970853ffe81112028962e1b/call.go#L150

**Comment by iamqizhao** [comment]
Thanks for reporting. I actually had a commit to fix this already which is
not checked in due to the lack of tests. Will add the tests and submit it
tomorrow.

**Comment by iamqizhao** [comment]
should be fixed in #132.  Please reopen it if the problem sustains.
